uitools/toolkitcpp/cpp/Esri/ArcGISRuntime/Toolkit/MediaPopupElementViewController.cpp

```diff
@@ -16,18 +16,81 @@
 #include "MediaPopupElementViewController.h"
 
 // Qt headers
-#include <QDebug>
+#include <QJsonObject>
+#include <QJsonDocument>
+#include <QJsonArray>
 
 // Maps SDK headers
 #include <MediaPopupElement.h>
 #include <PopupMedia.h>
 #include <PopupMediaListModel.h>
+#include <PopupTypes.h>
 
 // Toolkit headers
 #include <PopupMediaItem.h>
+#include <ImagePopupMediaItem.h>
+#include <BarChartPopupMediaItem.h>
+#include <PieChartPopupMediaItem.h>
+#include <LineChartPopupMediaItem.h>
 
 namespace Esri::ArcGISRuntime::Toolkit {
 
+namespace {
+
+  template <typename T>
+  bool isNullOrUndefined(const T& value) {
+      return value.isNull() || value.isUndefined();
+  }
```

Comment:
I don't think this is necessary to be templatized but I don't find the opportunity to do so very often so I went for it. 😄

Reply:
What types would you expect to come through here? It has to be something that supports both `isNull()` and `isUndefined()`.

Reply:
I thought more of the QJson... types had both but it looks like `isUndefined` is unique to QJsonValue. I'll curb my enthusiasm and reverse course as it's for the best 😄